BrowserKit/Tests/TabDataStoreTests/TabDataStoreTests.swift

```diff
@@ -58,37 +81,50 @@ final class TabDataStoreTests: XCTestCase {
                     DispatchQueue.main.async {
                         XCTAssertTrue(FileManager.default.fileExists(atPath: backupPath.path))
                     }
+                    do {
+                        let data = try Data(contentsOf: backupPath)
+                        let backupWindowData = try JSONDecoder().decode(WindowData.self, from: data)
+                        DispatchQueue.main.async {
+                            XCTAssertEqual(backupWindowData.id, windowData.id)
+                        }
+                    } catch {
+                        DispatchQueue.main.async {
+                            XCTFail("can't read the backup")
+                        }
+                    }
                 } else {
-                    XCTFail("can't create the backup path")
+                    DispatchQueue.main.async {
+                        XCTFail("can't create the backup path")
+                    }
                 }
```

Comment:
Why do these checks need to be on the main thread?
With swift concurrency if you want something to be on the main thread you can add @MainActor to the function defination, so this fucntion would be `@MainActor func testFetchBackup() async` , but in this case I'm not seeing why these checks need to be on the main thread.

Reply:
I was getting a weird crash when running the tests on the XCT assertions complaining that they must be called on the main thread, now it's not happening anymore so I'll get rid of those